### DISCUSSION TITLE
Setting up alerts

### DISCUSSION THREAD

**Discussion by Tthongsy** [datalust/seq-tickets#1337] [Help]
Hey,
I'm unsure if I set this alert correctly.
I'm trying to set up an alert to send an email when a oversized log seen.
Is this correct?
I'm also receiving a query timeout when trying to slice the dashboards by times, is there anyway around that?
![image](https://user-images.githubusercontent.com/88148249/136276857-6660681e-8038-4711-8395-c765c203e87e.png)

**Comment by nblumhardt** [comment]
Hi Tommy,

Assuming that you want to detect events that would otherwise be dropped because of the raw event payload limit, no - the events that are larger than the limit will never be ingested, and so the alert will never process them.

If you do something like the above, but increase the raw event payload limit and batch size limits to let the larger events in, this could work, but you would need to still maintain a reasonable payload size limit, and act to prevent/remove large events, to avoid running the risk of performance issues.

Tackling large event detection at the source (application-side) is the better approach, if you're able to make modifications there.

Regarding the dashboard timeout, sending some more details to `support@` with a screenshot, information about the chart in question (signals, columns, where clause, etc.) and a diagnostic report from _Settings > Diagnostics > Download a full diagnostic report_ should help us narrow down the problem/possible solutions.

Hope this helps!
Nick

**Reply by Tthongsy** [reply]
So I have it set that way with an increase raw event payload limit currently because there are only a few oversized logs but we want to catch them. 
I set up the dashboard to have an app send the email, its the emailPlus app.
I have it currently set up appropriately but I'm unsure if I should have "stream incoming events" ticked off or on because its technically already being done in alerts?
I essentially want alerts to be sent through email when an oversized event has gone through Seq
![image](https://user-images.githubusercontent.com/88148249/136314239-b79fa841-ba88-445a-b4a0-4fce5b174522.png)

**Comment by nblumhardt** [comment]
Ah, I see now! I missed a problem in your first screenshot: the `Length(@Document) > 262000` there would work as a `where` clause, but the alert condition is a `having` clause, where event properties like `@Document` aren't available.

Giving this some more thought, setting up a chart or signal based on `@Document` won't be great for performance. It's okay to use this for occasional investigations, but having `@Document` evaluated over everything in the stream all the time will put a strain on resources. `@Document` is a string with the entire event serialized as JSON, which needs to be dynamically computed.

Having said that, the best _direct_ option right now will be to set up a new signal with the filter `Length(@Document) > 262000`, choose _Stream incoming events_, and then choose the signal using the picker just below the checkbox. You won't need to set up a dashboard-based alert separately.

Seq 2021.3 includes some new alerting options that will be slightly better performance-wise, but, if it's at all possible, a client-side solution will be better in the long run.

Do you have the option to plug something into the client-side logging pipeline, e.g. an enricher? Or if we were to update the Seq sink with some additional facilities for logging these events, would you be able to roll it out as an update?

Best regards,
Nick

**Reply by Tthongsy** [reply]
Ahh that would make sense, I'm going to give _Stream incoming events_ a try and see if it works out. We are currently looking to upgrade all of our seq instances.
I should send a support email about that at some point, but currently we are trying to build a foundation.
I'll get back to you if these alerts aren't working.

Thanks for the help!

**Reply by Tthongsy** [reply]
I tried doing the Stream incoming events and I see events going through but I am not receiving any emails still. Do I have to set anything up through the console? Or is it all through the web portal?

**Reply by nblumhardt** [reply]
Hi! Should be all through the web portal.

When you say that you can see the events going through, you mean that beside the app name there's a number of events shown as being sent to the app?

If so, is there also a number next to the zigzag arrow showing diagnostics coming _from_ the app?

One quick way to test that an instance of the email app is working properly is to go to Events, click an event, and on the context menu choose Send to app > your app instance. If this doesn't work then the problem is likely to be with the app configuration (email server, ports, etc.)

If you can post screenshots of the various parts (the chart and alert configuration) here or to support@datalust.co, we might be able to spot something there, too.

HTH!

**Reply by Tthongsy** [reply]
![image](https://user-images.githubusercontent.com/88148249/137238557-2703ccb3-c9ed-48ff-b3e0-b08a347c60cc.png)

As far as I know all information is correct on our end. 

![image](https://user-images.githubusercontent.com/88148249/137238607-bc5d92ad-bea5-4ab0-b931-042257034c51.png)
This is after sending an event to the app.

Not too sure what could be wrong?

**Reply by nblumhardt** [reply]
Thanks for the extra info! I forgot you're running an older version, so the display of information for the app is different to what I described above. That blue hyperlinked double-down-arrow indicates that the app is emitting diagnostic errors, which will tell you why it's failing to send email.

If you click on the `↓1` then you should see the error messages being generated by the app.

**Reply by Tthongsy** [reply]
![image](https://user-images.githubusercontent.com/88148249/137251208-f5075a2b-8472-4def-a96a-15bb2fbdfa7b.png)

Thanks for that, I'm getting this error. Is this telling me that I have the incorrect port and other information?

**Reply by nblumhardt** [reply]
Great! This means that the name of the SMTP host machine ("Host", in your screenshot of the app configuration above) isn't a name that's visible from the Seq server.

If you open up a command prompt on the Seq server itself and type `nslookup <host>`, you should get some information about how the Seq server sees that DNS entry.

HTH!
